com.unity.mobile.notifications/Editor/AndroidNotificationPostProcessor.cs

```diff
@@ -15,11 +15,25 @@ public class AndroidNotificationPostProcessor : IPostGenerateGradleAndroidProjec
 
         public void OnPostGenerateGradleAndroidProject(string projectPath)
         {
+            MinSdkCheck();
+
             CopyNotificationResources(projectPath);
 
             InjectAndroidManifest(projectPath);
         }
 
+        private void MinSdkCheck()
+        {
+#if !UNITY_2020_2_OR_NEWER
+        // API level 21 not supported since 2020.2, need to check for prior releases
+        const AndroidSdkVersions kMinAndroidSdk = AndroidSdkVersions.AndroidApiLevel21;
+
+        if (PlayerSettings.Android.minSdkVersion < AndroidSdkVersions.AndroidApiLevel21)
+                throw new NotSupportedException(string.Format("Minimum Android API level supported by notifications package is {0}, your Player Settings have it set to {1}",
```

Comment:
maybe it's better to use https://docs.unity3d.com/ScriptReference/Build.IPreprocessBuildWithReport.OnPreprocessBuild.html, since it looks like you want to fail early, instead of making user wait for the whole build to finish and only then fail?

Reply:
Huh, if I throw exception out of that one, it logs it to console, but still continues with a build and produces apk!

Reply:
Huh, that's strange, maybe worth asking in #devs-asset-build-pipeline, sounds like a bug. I'll approve your PR, if this is blocking you

Reply:
Try BuildPlayerWindow.BuildMethodException

Reply:
Same issue. Maybe it's fixed in newer Unity, but I need this specifically for 19.4